### PR TITLE
Issue 7: Delete Image

### PR DESCRIPTION
**Pull request by hannaveldhuizen**
And finally, we can delete images!

### CODE REVIEW

**Comment by hannaveldhuizen** [comment]
[Issue 7: Delete Image](https://trello.com/c/cqBbQt3E/135-issue-7-delete-image)

File: app/views/images/index.html.erb [hunk 1]
```diff
@@ -14,6 +14,14 @@
             <a class='btn btn-light' href='<%= tags_path(tag) %>'><%= tag %></a>
           <% end %>
         </h3>
+        <%= button_to 'delete', {
+          :controller => :images,
+          :action => 'destroy',
+          :id => image.id
+        },
+          :method => :delete,
+          data: { confirm: 'Are you sure you want to delete this image?' }
```
Comment (fanniey) [inline app/views/images/index.html.erb]: 1. for hash key, we prefer symbol (`:key`) over string (`'key'`), i.e. `{ controller: :images }` over `{ :controller => :images }`. 

2. more info for `button_to`: https://api.rubyonrails.org/v5.2.3/classes/ActionView/Helpers/UrlHelper.html#method-i-button_to
The first input is the text on the button. 
The second input is for the url. 
The third input is for the html attributes. 
The second attribute can be replaced with rails url_helper method, `image_path(@image)`

3. the indention look a bit off here 
```
    <%= button_to 'delete', {
            :controller => :images,
            :action => 'destroy',
            :id => image.id
          },
          :method => :delete,
          data: { confirm: 'Are you sure you want to delete this image?' }
```

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -21,9 +23,12 @@ def show
     @image = Image.find(params[:id])
   end
 
-  def tags_show
-    @current_tag = params[:id]
-    @images = Image.tagged_with(@current_tag)
```
Comment (fanniey) [inline app/controllers/images_controller.rb]: I would split the commit into 
- Remove tag route 
- Add destroy action

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -95,16 +95,33 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+  test 'should delete the right image' do
+    img1 = Image.create(url: 'https://1.com')
+    img2 = Image.create(url: 'https://2.com')
+
+    assert_equal img2.id, Image.last.id
+    assert_difference 'Image.count', -1 do
+      delete image_url id: img2.id
+    end
+    assert_equal img1.id, Image.last.id
+  end
```
Comment (fanniey) [inline test/controllers/images_controller_test.rb]: I would also do a `assert_redirect_to images_path` here.

**Comment by fanniey** [comment]
I would prefer to make the commit history clean and the tests thorough. For the rope purposes, I think it is fine to merge this PR without addressing my last two comments if it has been too much back and forth for you.

**Pull request merged**